java/core/src/main/java/sleeper/core/statestore/AllFileReferences.java

```diff
@@ -16,29 +16,57 @@
 
 package sleeper.core.statestore;
 
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Stream;
 
 /**
  * This class contains a snapshot of files in the state store at a point in time, to be used to build a report.
  */
 public class AllFileReferences {
     private final Set<String> filesWithNoReferences;
-    private final Set<FileReference> activeFiles;
+    private final Map<String, FileReference> activeFilesByPartitionAndFilename;
+    private final Map<String, FileInfo> filesByFilename;
     private final boolean moreThanMax;
 
-    public AllFileReferences(Set<FileReference> activeFiles, Set<String> filesWithNoReferences, boolean moreThanMax) {
+    public AllFileReferences(Collection<FileReference> activeFiles, Set<String> filesWithNoReferences, boolean moreThanMax) {
```

Comment:
`activeFiles` would be better as `fileReferences`

Reply:
I've removed this constructor.